ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderLease.java

```diff
@@ -49,12 +49,12 @@ class LeaderLease {
         .toIntExact(TimeUnit.MILLISECONDS);
   }
 
-  boolean isEnabled() {
-    return enabled;
+  boolean getAndSetEnabled(boolean enabled) {
```

Comment:
when you set this field to true?

It looks like now it won't be true as long as it's set to false

Reply:
It will be set back to original value if we failed transferring leadership. In other situations (stepDown / successful transferring), the LeaderStateImpl will be stopped and the `enabled` will stay in false.